src/engines/spidermonkey.js

```diff
@@ -32,23 +32,38 @@ class SpiderMonkeyInstaller extends Installer {
 
   static async resolveVersion(version) {
     if (version === 'latest') {
-      const data = await fetch('https://product-details.mozilla.org/1.0/firefox_history_development_releases.json')
-        .then((r) => r.json());
-      let latestVersion = 0;
-      let latestTimestamp = 0;
-      for (const [key, value] of Object.entries(data)) {
-        const timestamp = +new Date(value);
-        if (latestTimestamp < timestamp) {
-          latestTimestamp = timestamp;
-          latestVersion = key;
+      // Build a request for buildhub2: https://buildhub2.readthedocs.io/en/latest/project.html
+      const body = {
+        size: 1,
+        sort: {'build.id': 'desc'},
+        query: {
+          bool: {
+            must: [
+              {term: {'source.product': 'firefox'}},
+              {term: {'source.tree': 'mozilla-central'}},
+              {term: {'target.channel': 'nightly'}},
+              {term: {'target.platform': getFilename()}},
+            ]
+          }
         }
-      }
-      return latestVersion;
+      };
+
+      const data = await fetch('https://buildhub.moz.tools/api/search', {
+        method: 'post',
+        body: JSON.stringify(body),
+      }).then((r) => r.json());
+
+      const source = data.hits.hits[0]._source;
+
+      return `${source.target.version}@${source.build.id}`;
     }
     return version;
   }
 
   getDownloadURL(version) {
+    if (this.isLatest) {
```

Comment:
i don't think it should be possible to automatically install versions that can't be installed manually

Reply:
I can change the PR to use a build-id specific download URL, like <https://archive.mozilla.org/pub/firefox/nightly/2020/03/2020-03-25-09-34-57-mozilla-central/>, if you think that's preferable.

Reply:
do nightly and beta builds have noticeably different version specifiers? if there's some pattern to check by includes() or regex or something that would be fine.

Reply:
Kind of. To uniquely identity a Nightly version, it's necessary to include a build-id. All Nightlies for the current release cycle will have the version number "76.0a1", so when requesting a Nightly from a specific date, the build-id needs to be included, too. For example the latest Nightly has the [build-id "20200325093457"](https://buildhub.moz.tools/?q=20200325093457). (`about:support` can be used to display the build-id in Firefox.) 

The current PR appends the build-id after the version identifier in `resolveVersion()`, for example "76.0a1@20200325093457". `getDownloadURL()` could then check for this pattern to compute the correct download location.

Reply:
Looks like I can't use `"@"` as a delimiter for the build-id, because it breaks the `parseEngine()` function, so I've changed it to `"#"` for now.